pkg/option/config.go

```diff
@@ -2834,8 +2831,9 @@ func (c *DaemonConfig) Populate(vp *viper.Viper) {
 	c.EnableXDPPrefilter = vp.GetBool(EnableXDPPrefilter)
 	c.EnableTCX = vp.GetBool(EnableTCX)
 	c.DisableCiliumEndpointCRD = vp.GetBool(DisableCiliumEndpointCRDName)
-	c.MasqueradeInterfaces = vp.GetStringSlice(MasqueradeInterfaces)
-	c.EgressMasqueradeInterfaces = strings.Join(c.MasqueradeInterfaces, ",")
+	if masqInfs := vp.GetString(MasqueradeInterfaces); masqInfs != "" {
+		c.MasqueradeInterfaces = strings.Split(masqInfs, ",")
+	}
```

Comment:
Along with a revert of the flag type change, I think this should be sufficient:

```suggestion
c.EgressMasqueradeInterfaces = c.MasqueradeInterfaces
```

Then the only remaining question is whether the documentation examples need to be updated to make it clear how to configure multiple options.

Reply:
Sorry but I'm not following this...
`EgressMasqueradeInterfaces` and `MasqueradeInterfaces` have different types.
Also, `EgressMasqueradeInterfaces` is not used anywhere.

Reply:
Ah sorry, I sent this in a bit of a hurry. I meant this:

```suggestion
       c.MasqueradeInterfaces = vp.GetStringSlice(MasqueradeInterfaces)
```

Reply:
Regarding default value for the `MasqueradeInterfaces` param, I found the following two places in the code:
1. https://github.com/cilium/cilium/blob/main/daemon/cmd/daemon.go#L653
2. https://github.com/cilium/cilium/blob/main/pkg/datapath/iptables/iptables.go#L1198

So, they might be impacted by default value, WDYT?

Reply:
I think that's the idea yes.